include/ignition/rendering/base/BaseArrowVisual.hh

```diff
@@ -62,23 +84,56 @@ namespace ignition
     {
     }
 
+    /////////////////////////////////////////////////
+    template <class T>
+    void BaseArrowVisual<T>::Destroy()
+    {
+      while (this->ChildCount() > 0u)
```

Comment:
The child count is decremented every time `Destroy()` is called on one of its children.
Doing this step is necessary because an arrow visual associated with a joint still remains in the scene after removing the model:
![Screenshot from 2021-08-18 11-05-10](https://user-images.githubusercontent.com/39728574/129843411-48cae9ab-5951-4183-8362-e2718528a893.png) ![Screenshot from 2021-08-18 11-05-23](https://user-images.githubusercontent.com/39728574/129843419-4a112343-ef28-4afd-9298-95674f7e0552.png)
This is similar to what I faced in the `InertiaVisual` class, the purple box remained in the scene even after destroying its parent visual (Inertia).

Maybe we can, to be safe, store the references to arrow visual children (shaft, head, and rotation) explicitly in variables, and destroy them whenever `ArrowVisual->Destroy()` is called?

Currently, arrow shaft, rotation, and head are accessed by index:
https://github.com/ignitionrobotics/ign-rendering/blob/4884de6cc4c0b798d9214101f53b726fa73503fe/include/ignition/rendering/base/BaseArrowVisual.hh#L103-L120

Reply:
ok sounds good. I think this is fine. Can you add this snippet below to the end of `ArrowVisual_TEST.cc` so we have some coverage for this bit of code?

```
  // test destroy
  ArrowVisualPtr visual2 = scene->CreateArrowVisual();
  ASSERT_NE(nullptr, visual2);
  visual2->Destroy();
  EXPECT_EQ(0u, visual2->ChildCount());
```

Reply:
Added in [`fc6c6e3`](https://github.com/ignitionrobotics/ign-rendering/pull/366/commits/fc6c6e328e9cda836cacdae5a439895fed9d5153)